src/otx/algorithms/common/adapters/torch/amp/xpu_grad_scaler.py

```diff
@@ -0,0 +1,117 @@
+"""Custom GradScaler to scale loss."""
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+from collections import abc, defaultdict
+from typing import List
+
+import intel_extension_for_pytorch as ipex  # noqa: F401
+import torch
+from intel_extension_for_pytorch.cpu.autocast._grad_scaler import _MultiDeviceReplicator
```

Comment:
Same as above, if difference between `XPUGradScaler` and original `GradScaler` is only `_MultiDeviceReplicator`, then it can be applied as code below
```
if is_xpu_available():
    from intel_extension_for_pytorch.cpu.autocast._grad_scaler import _MultiDeviceReplicator
    from torch.cuda.amp import grad_scaler
    grad_scaler._MultiDeviceReplicator = _MultiDeviceReplicator
```

Reply:
There are more difference, such as
https://github.com/openvinotoolkit/training_extensions/pull/2668#discussion_r1405597556
https://github.com/openvinotoolkit/training_extensions/pull/2668#discussion_r1405597665